### ISSUE TITLE
[HOLD for payment 2024-04-15] [$250] HIGH: [Comment linking] Highlight the linked comment with a different color than the normal hover color

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#38851]
https://expensify.slack.com/archives/C035J5C9FAP/p1711152608943449?thread_ts=1705035404.136629&cid=C035J5C9FAP

## Action Performed:
Click on a link to a comment in any chat.

## Expected Result:
1. You should be linked straight to that comment and be able to load chats in either direction (already built, this is currently on staging)
1. The comment you linked to should be highlighted in a different color than the normal hover color (temporarily? or permanently?)

## Actual Result:
The linked-to comment is highlighted with the color that a message becomes when you hover over it.

## Workaround:
n/a

## Platforms:
all platforms

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d8f674d9f769e10c</li>
        <li>Upwork Job ID: 1774424743522938880</li>
        <li>Last Price Increase: 2024-03-31</li>
<li>Automatic offers: </li>
<ul>
<li>Pujan92 | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Pujan92</details>

**Comment by roryabraham** [comment]
hey @shawnborton, any thoughts on what color we could use to highlight the message that you linked to?

**Comment by quinthar** [comment]
ya the comment linking worked great!  However, I initially didn't realize what comment I had linked to, because it just looked like I was hovering over it:
![image](https://github.com/Expensify/App/assets/575186/c5fec42a-48bb-4b78-a806-df76563b3acd)

**Comment by quinthar** [comment]
I would think it'd remain permanently highlighted (unlike Slack), until the page changes.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
- In [here](https://github.com/Expensify/App/blob/2003844351cd57b5197852459bc36ec07b6a7d26/src/pages/home/report/ReportActionItem.tsx#L186), we apply ```theme.hoverComponentBG``` if the report action is linked report action
### What changes do you think we should make in order to solve the problem?
- We can replace ```theme.hoverComponentBG``` by our expected color code

### What alternative solutions did you explore? (Optional)

- NA

**Comment by kadiealexander** [comment]
@shawnborton your thoughts on the highlight colour would be great, thanks!

**Comment by shawnborton** [comment]
Yeah, I think we should just nudge this one up to use our border color as the highlight color - it's consistent with how we highlight active nav items in the LHN too. And I like the idea to leave it highlighted until you navigate away. 
![CleanShot 2024-03-25 at 11 56 01@2x](https://github.com/Expensify/App/assets/2319350/3ed91a13-1c5a-4f71-8c5b-bc58b0c000f0)

We could try other colors if we wanted to, like a shade of yellow or something. Thoughts? cc @Expensify/design 

![CleanShot 2024-03-25 at 12 00 30@2x](https://github.com/Expensify/App/assets/2319350/d49f3d7f-94d0-4161-b30e-30a92349c2f9)

**Comment by dannymcclain** [comment]
Oh the yellow is kinda interesting! Can you point me to the Figma file? I'd love to see your border color example with a hovered message as well, just to compare. But I'm definitely intrigued by the yellow...

**Comment by shawnborton** [comment]
Threw it in the ol' dumpster [here](https://www.figma.com/file/WlrZyRCTQ0LFoKcjSmtTsd/General-UI-Patterns?type=design&node-id=3395%3A10&mode=design&t=letXKbu8pW2WDI4e-1)

**Comment by roryabraham** [comment]
I'm a fan of the yellow as well 💛

**Comment by dubielzyk-expensify** [comment]
Yellow feels appropriate if we need to go away from the green hue. All the others seem to harsh except for something like `ice`.

**Comment by quinthar** [comment]
What is the next step here, who is doing it, and what is the ETA?

**Comment by nkdengineer** [comment]
Can the contributor work on it? If can, anyone can review my proposal [here](https://github.com/Expensify/App/issues/38851#issuecomment-2016429089). Thanks

**Comment by shawnborton** [comment]
Any more feedback on the colors @Expensify/design? Feel free to explore some ideas if you have any!

**Comment by dannymcclain** [comment]
@shawnborton I explored a few but honestly I'm pretty into the yellow haha. 💛

The yellow is very clearly _different_ from "normal" product colors, so it makes it stand out as a highlight vs. a hover/other UI state. Which I think makes it a lot more clear what's happening. I don't know that I'm 100% sold on the highlight staying "forever" (until the page changes) but I feel like if we wanted to revisit the timing, we could do that as a follow-up. 

The only design-y thing to note is that we should define these colors as product variables. (Light mode uses `yellow-100` but dark mode uses an opacity of a yellow swatch, so I'd like to have that locked down to a variable for easiness.)

**Comment by dubielzyk-expensify** [comment]
I'm with y'all on the yellow. The fact that yellow is a highlighter color is too serendipitous as well. 

Good shout @dannymcclain . I could see us adding a `highlight` color in Figma.

**Comment by dubielzyk-expensify** [comment]
Okay, I used it in context and I do love it on the light mode, but I think the dark mode highlight looks pretty muddy so I shift my vote to the blue.

**Comment by shawnborton** [comment]
Yeah, I totally agree the yellow feels muddy when we use reduced opacity like that. 

One thing we can try to do is to use true color values from our system, and don't use any opacity. After a few attemps, I think yellow (800/100) or ice (800/100) could work well for dark/light respectively. Of these two, I am digging ice quite a bit:
![CleanShot 2024-03-28 at 11 40 56@2x](https://github.com/Expensify/App/assets/2319350/290b0ced-e81c-4e39-8a52-c051143b4ec2)


I tried other combinations that didn't quite work as well:
![CleanShot 2024-03-28 at 11 42 46@2x](https://github.com/Expensify/App/assets/2319350/9394c07f-ace6-41ff-bf9f-7a47e9c7ccf6)

**Comment by quinthar** [comment]
I think Yellow is the most commonly used for highlighting in this situation so will be the most familiar, I vote Yellow.

**Comment by shawnborton** [comment]
Cool, if we do that, then the values I proposed above to stay true to our color system: 
![image](https://github.com/Expensify/App/assets/2319350/a22e20f5-1a8f-46b4-ad29-6d905ef4fcde)


Would be:
- `yellow100` - `#FFF2B2`
- `yellow800` - `#401102`

Definitely curious for @Expensify/design's thoughts though when they are online.

**Comment by dannymcclain** [comment]
I'm totally down to roll with the yellows, but in contrast to Jon's opinion, I actually like the original yellow-with-opacity more than `yellow-800` (even though I agree it does feel a _little_ muddy). I don't feel strongly enough to delay moving forward though—I just like how the opacity feels a bit more "yellow" than the `800` swatch.

Ice is also nice, but on dark mode it starts to feel more like a "regular" UI color.
Blue also looks really good, but doesn't have the same "highlight" vibe as yellow.
And of course green looks great too, but I worry it doesn't stand out quite as much as the yellow.

So all that to say, I'm down with yellow but would also be pretty happy with any of the options (green would probably be my top second pick).

**Comment by shawnborton** [comment]
Yeah, I think that is totally fair feedback. Slack ends up doing something quite muddy on dark mode too for reference:
![CleanShot 2024-03-29 at 14 40 32@2x](https://github.com/Expensify/App/assets/2319350/54cc4e74-18c2-4860-ad2b-458978c165f4)


So we could just take our `yellow800` and push the hue more towards traditional yellow, and we get something like this:
![CleanShot 2024-03-29 at 14 40 55@2x](https://github.com/Expensify/App/assets/2319350/fa63e542-e837-488c-8618-35443c627e6f)

Basically just moving this knob from left to right a bit:
![CleanShot 2024-03-29 at 14 41 16@2x](https://github.com/Expensify/App/assets/2319350/55db7ff2-c1c1-45c6-a718-5d1695a64acf)

**Comment by dannymcclain** [comment]
I'm down for that 👍

**Comment by shawnborton** [comment]
Cool, and it seems like something that we can easily change, even during the PR review process, once we actually take it for a spin. 

So the colors to start:
- Dark mode highlight: `#402B02` - a new custom color
- Light mode highlight: `#FFF2B2` - aka `yellow100`

**Comment by dannymcclain** [comment]
Sounds good!

>I could see us adding a highlight color in Figma.

I think we should go ahead and add `message highlight` variables in our `surface` group of our Product:Colors variables. That way if we do decide to change it, it will be _very_ easy.

**Comment by shawnborton** [comment]
Sweet - I just added it but I added it to state as it seemed to be more appropriate there?
![CleanShot 2024-03-29 at 17 19 53@2x](https://github.com/Expensify/App/assets/2319350/28029f08-4d87-4fa5-8701-03ccab21b4a4)


Mostly because highlighted is a state that can change, so that was my rough thinking. Thoughts?

**Comment by dannymcclain** [comment]
Yeah I like that! We might want to name it something else though—just because we already have a `highlight-background`. What about just `state/highlight`?

**Comment by shawnborton** [comment]
Oh that's a good shout - yup, let's go with what you suggested. Will update!

**Comment by quinthar** [comment]
Ok, what is the next step -- who is building this?  Do we have a PR?

**Comment by nkdengineer** [comment]
I updated the screenshot videos in [proposal](https://github.com/Expensify/App/issues/38851#issuecomment-2016429089) based on this [comment](https://github.com/Expensify/App/issues/38851#issuecomment-2027380607)

**Comment by TafreedAhmad** [comment]
Contributor details
Your Expensify account email: atafreed1@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01b56ff0baf8336058

**Comment by Pujan92** [comment]
> should i try to do it or is someone else onto it?

@TafreedAhmad For this @nkdengineer is assigned and they will do the task. You can look out the other issues where **Help Wanted** label is applied.

**Comment by quinthar** [comment]
@nkdengineer what is your ETA for completing this?  I assume this is very fast, can you finish it today?

**Comment by nkdengineer** [comment]
@Pujan92 The PR is [here](https://github.com/Expensify/App/pull/39346).

**Comment by nkdengineer** [comment]
@quinthar Yes, I can finish it today.

**Comment by kadiealexander** [comment]
Payouts due:

- [ ] Contributor: $250 @nkdengineer
- [ ] Contributor+: $250 @Pujan92

Upwork job is [here](https://www.upwork.com/ab/applicants/1774424743522938880/job-details).

**Comment by kadiealexander** [comment]
@Pujan92 please propose the regression test steps

**Comment by Pujan92** [comment]
### Regression Test Proposal

1. Copy link for any of the report message with the context menu
2. Open that link
3. Verify that the linked message is highlighted

Do we agree 👍 or 👎
